ocaml/xapi/xapi_host.ml

```diff
@@ -518,6 +518,40 @@ let reboot ~__context ~host =
   shutdown_and_reboot_common ~__context ~host "Host is rebooting" "Host is rebooting"
 	`shutdown "/sbin/shutdown -r now"
 
+let read_lines fi =
+	let rec loop acc =
+		let l = try Some (input_line fi) with _ -> None
+		in match l with
+			| None -> List.rev acc
+			| Some l -> loop (l :: acc)
+	in loop []
+
+let read_file file =
+	let fi = try Some (open_in file) with _ -> None in
+	match fi with
+	| None -> []
+	| Some fi ->
+		let lines = read_lines fi in
+		close_in fi ;
+		lines
+
+let rec is_empty_dir dir =
```

Comment:
If the purpose of this function to just check that the directory is empty, this should suffice:

    let is_empty_dir dir = Array.length (Sys.readdir dir) = 0

If the purpose is something else, you should change the name.